### PR TITLE
board: arm: google_kukui: Add a sample for kukui

### PR DESCRIPTION
**Pull request by sjg20**
It is useful to use kukui with a shell to test out its features., Add a
sample for this.

Signed-off-by: Simon Glass <sjg@chromium.org>

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
Is it WIP? Where is the essential thing that this sample produces for this board? Which board would not work with it?

**Comment by sglass68** [comment]
> Is it WIP? Where is the essential thing that this sample produces for this board? Which board would not work with it?
> 
> And why not just use `samples/subsys/shell/shell_module`?

It is WIP but it is a start. With this it is possible to try things out as development continues. I was told not to add shell support to a board.

The shell_module example does not enable shells for all commands though.

**Comment by carlescufi** [comment]
> The shell_module example does not enable shells for all commands though.

Is this something we could fix by having a `boards/google_kukui.conf` in that sample?

**Comment by sglass68** [comment]
Possibly. This is intended to be an EC (Embedded Controller for laptop/tablet) implementation. I am happy to put it anywhere that suits since we can always move it later.

**Comment by carlescufi** [comment]
> Possibly. This is intended to be an EC (Embedded Controller for laptop/tablet) implementation. I am happy to put it anywhere that suits since we can always move it later.

I see, so this is a bit more than a sample, you intend to have a full implementation of an EC in-tree. I am not opposed to this, but would like to hear other opinions.

BTW, this is a new sample so it cannot go in before 2.3.0 is tagged unless you label it as TSC and bring it to the TSC meeting.

**Comment by sglass68** [comment]
I am not sure about 'full implementation' right now, or where exactly it might end up. But definitely I think adding the drivers and seeing where we end up.

There is no urgency with this so it can wait until after the release.

**Comment by sjg20** [comment]
Can I get another look at this one please?

File: samples/boards/google_kukui/README.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.. _google_kukui:
+
+Kukui general features
+######################
+
+Overview
+********
+
+This provides access to Kukui through a serial shell so you can try out the
```
Comment (mbolivar-nordic) [inline samples/boards/google_kukui/README.rst:9]:
```suggestion
This provides access to :ref:`Kukui <google_kukui_board>` through a serial shell so you can try out the
```

File: samples/boards/google_kukui/README.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   :board: google_kukui
+   :goals: build
+   :compact:
+
+Sample Output
+=============
+
+.. code-block:: console
+
+    Hello World! google_kukui
```
Comment (mbolivar-nordic) [inline samples/boards/google_kukui/README.rst:28]:
```suggestion
    Welcome to Google Kukui
```

**Review by mbolivar-nordic — COMMENTED** [review]
A couple of drive-by docs suggestions

**Review by mbolivar-nordic — APPROVED** [review]
Seems reasonable to me if it is useful to an actively maintained board's users.

**Review by nashif — APPROVED** [review]

**Pull request merged**